pkg/lorry/plugin/proto/db_plugin.proto

```diff
@@ -0,0 +1,256 @@
+
+  // The port on which the DB service listens.
+  string port = 2;
+
+  // The admin username used to access the DB service.
+  string admin_user = 3;
+
+  // The admin password of the user used to access the DB service.
+  string admin_password = 4 [(kb_secret) = true];
+}
+
+message GetRoleRequest {
+  DBInfo db_info = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message GetRoleResponse {
+  string role = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message JoinMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the new member's Pod.
+  string new_member = 5;
+
+  // The IP of the new member's Pod.
+  string new_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message JoinMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message LeaveMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the leave member's Pod.
+  string leave_member = 5;
+
+  // The IP of the leave member's Pod.
+  string leave_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message LeaveMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadonlyRequest {
+  DBInfo db_info = 1;
+
+  // The reason for setting the replica as read-only.
+  string Reason = 2;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 3;
+}
+
+message ReadonlyResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadwriteRequest {
+  DBInfo db_info = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message ReadwriteResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message AccountProvisionRequest {
+  DBInfo db_info = 1;
+
+  // the user name of the new account.
+  string user_name = 2;
+
+  // the password of the new account.
+  string password = 3 [(kb_secret) = true];
+
+  // the role of the new account.
+  string role = 4;
+
+  // Common metadata property for extention
+}
+
+message AccountProvisionResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message SwitchoverRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica.
+  string primary_fqdn = 2;
```

Comment:
current old primary?

Reply:
Yes, this field is removed as it is duplicated with serviceinfo.fqdn.